### ISSUE TITLE
Resample parameter in sct_straighten_spinalcord is confusing

### ISSUE THREAD

**Issue by benjamindeleener** [neuropoly/spinalcordtoolbox#1433]
The `-resample` parameter in the documentation of `sct_straighten_spinalcord` is confusing:

```
 -resample <float>            Isotropic resolution of the straightening output, in millimeters.
                                Resampling to lower resolution decreases computational time while 
                                  decreasing straightening accuracy.
                                To keep native resolution, set this option to 0.
```

It suggests that the resolution of the straightening output is changed to the isotropic resolution passed to the function, while it only changes the resolution of the native image to increase/decrease the accuracy and computational time of the straightening process while keeping the native resolution at the end.

**Question:**
Should we allow the user to choose a resolution for the straightening space when adding this parameter?

**Comment by jcohenadad** [comment]
not sure i understand the current behavior: let's take this example: if user enters `-resample 1x1x1`, will it produce a straightened spinal cord at 1x1x1mm resolution?

**Comment by benjamindeleener** [comment]
No it will not. It will only change the computational time. I remember having a discussion with you about this, but I don't really remember why we decided to this like this. That's why I would like to have your opinion. Should we enable this feature?

**Comment by jcohenadad** [comment]
hum, indeed that's not very intuitive.  A few questions for you:
1) Is the output resolution the same as the native resolution? 
2) Is it currently possible to set the output resolution?
3) what are the disadvantages (apart from accuracy) of running the straightening with subsampling (while keeping the same output resolution as the input)? Did we run into issues such as revere warping field that cannot be properly concatenated, or tricky things like that? I remember this was a problem in the past...

**Comment by benjamindeleener** [comment]
@jcohenadad 

1. yes, the output resolution is currently always the same as the native resolution
2. no
3. except if going into large resolution (like 2x2x2mm or 3x3x3mm), I don't remember seeing any accuracy issues. You could also increase the resolution but it would basically be similar to an interpolation and it would take much more time.

If we are to allow the output resolution, I would suggest allowing non-isotropic resolution as well.

**Comment by jcohenadad** [comment]
1) are there currently some applications were we would like set the output resolution?
2) yes, i agree we should allow to set non-isotropic resolution (another argument is that, as you saw, i misinterpreted the flag and wrote 1x1x1, which i find more intuitive)

**Comment by charleygros** [comment]
are there currently some applications were we would like set the output resolution?
--> Project Lesion automatic segmentation:
I would like to straight the cord of all my subjects and impose a common resolution (multi-center subjects). That's why I was thinking to use the flag `-resample`.
(but the combination `sct_straighten_spinalcord` and then `sct_resample` could do the job : ) )

**Comment by jcohenadad** [comment]
@charleygros why not registering to the PAM50 directly? this will give you common resolution and common space

**Comment by charleygros** [comment]
@jcohenadad : Yes absolutely --> it is another solution I thought! (Followed by cropping to work with smaller images)

I talked to Ben about the `-resample` flag because reading the documentation I expected the output to be resampled.

**Comment by jcohenadad** [comment]
I think we should wither rephrase the flag (and keep current functionality), or change the output resolution. I really have no strong opinion about this. 
@benjamindeleener in the futur isn't the plan to have sct_straighten_spinalcord also deal with non-linear adustment along Z, to match vertebral levels of destination image? In that case we could also imagine a flag -d to match resolution (and possibly levels) of destination.

**Comment by benjamindeleener** [comment]
@jcohenadad Registering two images is actually already possible with the `-ref` parameter (which should actually be called `-d` for destination, it is a centerline image). Adjusting for vertebral levels requires providing binary label images with the position of intervertebral disks (`-disks-input` and `-disks-ref`).

I have no strong opinion either but I would suggest this:
1. parameter `-resample` (ex: 1x1x1, in mm) sets the resolution of the output straight space
2. parameter `-d` or `-dest` for registering the input image to another image (but it should be a centerline image), based on the length of the spinal cord (centerline must start at the same anatomical point to work). If providing the disks, the vertebral levels are adjusted.

**Comment by jcohenadad** [comment]
agreed with the philosophy, but i would like to get more consensus about the flags (ref vs. dest, differenciating "d" between image and centerline, etc.). We need more discussion on this.

**Comment by jcohenadad** [comment]
Following consensus during SCT meeting, the list of changes are:
- remove the flag "resample"
- add a flag "speed_factor" 
- add a flag "xy_size" which specifies in voxels/mm the dimension of the output image
- document the flags: ref, disc-input, disc-ref, give an example, and maybe point to an online documentation about how to do disc labeling.

**Comment by jcohenadad** [comment]
Fixed in #1669
